### PR TITLE
i18n(zh-CN): cleanup `guides\content-collections.mdx`

### PR DESCRIPTION
**Pull request by Genteure**
#### Description (required)

Cleanup and improve `guides\content-collections.mdx`.

- Add missing sentences.
- Update outdated code blocks.
- Remove duplicated paragraphs.
- Minor rewording.
- Formatting changes.

### CODE REVIEW

File: src/content/docs/zh-cn/guides/content-collections.mdx [hunk 1]
```diff
@@ -1,7 +1,8 @@
 ---
 title: 内容集合
 description: >-
-  内容集合有助于组织你的 Markdown 文件以及通过模式使用带有类型检查的 frontmatter。
+  内容集合有助于组织你的 Markdown 文件以及通过 schema 使用带有类型检查的 frontmatter。
```
Comment (100gle) [inline src/content/docs/zh-cn/guides/content-collections.mdx:4]: Are you sure that keep "schema" without translation? If that, maybe you also need to add this word to terms of translation in [Chinese translation guide](https://github.com/withastro/docs/tree/main/src/i18n/zh-cn) in next PR. 🤔

**Review by 100gle — APPROVED** [review]
LGTM

**Comment by Genteure** [comment]
I reverted change to the heading so links from other pages still work as expected. We can change this heading later.

In particular `reference/api-reference.mdx` is out of date for 3 weeks with 6 commits changing the source. Let's get this merged first.

**Review by Yan-Thomas — APPROVED** [review]
LGTM! Good work peeps 🙌

**Pull request merged**